clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3655,6 +3663,14 @@ LangAS CodeGenModule::getStringLiteralAddressSpace() const {
   // OpenCL v1.2 s6.5.3: a string literal is in the constant address space.
   if (LangOpts.OpenCL)
     return LangAS::opencl_constant;
+  if (LangOpts.SYCLIsDevice && !getenv("DISABLE_INFER_AS"))
+    // If we keep a literal string in constant address space, the following code
+    // becomes illegal:
+    //
+    //   const char *getLiteral() n{
+    //     return "AB";
+    //   }
+    return LangAS::opencl_private;
```

Comment:
```
    // If we keep a literal string in constant address space, the following code
    // becomes illegal:
```
Why not global then?

Reply:
This is a good question - I don't know what is the best address space for this.
I used private address space mainly because it was used to lower a string literal before I changed address spaces by this patch.